src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,379 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Metafunctions.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/FaceNormal.hpp"
+#include "Domain/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/NormalDotFlux.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t>
+class Mesh;
+namespace Tags {
+template <typename>
+struct NormalDotFlux;
+template <typename>
+struct Normalized;
+}  // namespace Tags
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace elliptic {
+namespace dg {
+namespace NumericalFluxes {
+
+/*!
+ * \brief The internal penalty flux for first-order elliptic equations.
+ *
+ * \details Computes the internal penalty numerical flux (see e.g.
+ * \cite HesthavenWarburton, section 7.2) dotted with the interface unit normal.
+ *
+ * We implement here a suggested generalization of the internal penalty flux
+ * for any set of elliptic PDEs. It is designed for fluxes (i.e. principal parts
+ * of the PDEs) that may depend on the dynamic fields. This is the case for the
+ * velocity potential equation of binary neutron stars, for example.
+ *
+ * Formulating the elliptic PDEs in terms of a first-order _auxiliary_ variable
+ * \f$v\f$ for each _primal_ variable \f$u\f$, they take the _flux-form_
```

Comment:
Are auxiliary variables always just first derivatives of the primal variables? 

Would it be clearer her to say "Reducing the second-order elliptic PDEs to first-order form by introducing an auxiliary variable for each primal variable (i.e., for each variable whose second derivative appears in the equations)."?

Reply:
- No, they can be any linear function of the primal variables and their derivatives. For example, for the XCTS system I'm currently working with the _symmetrized_ shift gradient to reduce the number of variables that are being solved for.
- Adjusted the text in a fixup